### ISSUE TITLE
No need for "typedef" for (declaration + assignment)

### ISSUE THREAD

**Issue by daletune** [palantir/tslint#669]
Please, tell me is it possible to get have this behavior or linter?

When typedef ("variable-declaration" or "member-valiable-declaration") options specified we have
warnings for both cases.

``` typescript
//1)
var myNumber;
myNumber = 2;
//2)
var myNumber = 2;
```

but we wanna warning only for first case,  because in the second scenario compiler knew the type from assignment.

Also this is applied to class variables.

``` typescript
class MyClass {
    public mySomething = 2;
}
```

Basically, the pattern is.
If assignment here, and assignment != null;

So, is there any solution or maybe I miss something?
Thank you.

**Comment by adidahiya** [comment]
In general, given the spirit of how you're using the `typedef` rule, it sounds like the `--noImplicitAny` compiler option should suffice...

The `typedef` rule doesn't do any inference right now, but I'll leave this issue open for discussion if there is sufficient need for some different behavior.

**Comment by daletune** [comment]
Thank you, @adidahiya. ```--noImplicitAny``` actually the thing we wanny use. But we have a big codebase so not all of those code could be build with flag for now. We have tslint check-in policy in our VCS so newer changes to files validated (so old ones not checked if you don't change them). That's the reason for this issue.

**Comment by robertbaker** [comment]
+1 the type is inferred by the assignment. The TSC and VS Code intellisense is smart enough to know the type from the assignment type.

**Comment by adidahiya** [comment]
@robertbaker is your use case the same as above? I want to avoid duplicating functionality of the compiler. Does `--noImplicitAny` not satisfy your use case?

**Comment by robertbaker** [comment]
That option would work, I don't have it on because didn't want to fix the errors yet.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
